bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/console/FirmwareUpdateConsoleCommandExtension.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2016 Deutsche Telekom AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.thing.internal.console;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingRegistry;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.firmware.Firmware;
+import org.eclipse.smarthome.core.thing.firmware.FirmwareRegistry;
+import org.eclipse.smarthome.core.thing.firmware.FirmwareStatus;
+import org.eclipse.smarthome.core.thing.firmware.FirmwareUID;
+import org.eclipse.smarthome.core.thing.firmware.FirmwareUpdateService;
+import org.eclipse.smarthome.io.console.Console;
+import org.eclipse.smarthome.io.console.extensions.AbstractConsoleCommandExtension;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * {@link FirmwareUpdateConsoleCommandExtension} provides console commands for managing the firmwares of things.
+ *
+ * @author Thomas Höfer - Initial contribution
+ */
+public final class FirmwareUpdateConsoleCommandExtension extends AbstractConsoleCommandExtension {
+
+    private static final String SUBCMD_LIST = "list";
+    private static final String SUBCMD_STATUS = "status";
+    private static final String SUBCMD_UPDATE = "update";
+
+    private static final List<String> ALL_SUBCMDS = new ImmutableList.Builder<String>().add(SUBCMD_LIST)
+            .add(SUBCMD_STATUS).add(SUBCMD_UPDATE).build();
+
+    private FirmwareUpdateService firmwareUpdateService;
+    private FirmwareRegistry firmwareRegistry;
+    private ThingRegistry thingRegistry;
+
+    public FirmwareUpdateConsoleCommandExtension() {
+        super("firmware", "Manage your thing´s firmwares.");
+    }
+
+    @Override
+    public void execute(String[] args, Console console) {
+        int numberOfArguments = args.length;
+        if (numberOfArguments < 1) {
+            return;

```

Comment:
IMHO we should do _something_ here. As there is no obvious default action, how about printing some help then?